### ISSUE TITLE
develop testing infrastructure for IDE integration

### ISSUE THREAD

**Issue by xeno-by** [scalameta/paradise#5]
It is important to make sure that new-style ("inline") macros work reasonably well with IDEs from the very beginning. I'd like to invite everyone to comment on what/how we should test to control the process and measure the progress.

**Comment by fommil** [comment]
absolutely! I wanted to create something in https://github.com/fommil/imaginary-friend but haven't created the framework yet. My goal was to release a test framework for macro authors to use to make sure their code works in ENSIME and Scala IDE and / or encourage people to prefer compiler plugins for more radical transforms.

However, I'm feeling a bit burnt out from heavy work on ENSIME these last 6 months (maybe 2 years?) so I'm probably going to take a break for a few months after we release 1.0... but I'm keen to get involved in this (when I can find the time).

**Comment by fommil** [comment]
btw, what about @pavelfatin ? (incidentally, we should introduce you to our GSoC student who's working on what the Concrete Syntax Trees are morphing into. I'd really like to see if IntelliJ could use at least some of that API so that we can generally share more code)

**Comment by fommil** [comment]
also @sschaef
